Final edits before publishing

cloudhead opened the pull request.

abbey-titcomb commented:
Suggested changes in _**bold italic**_

>> Throughout the last decade, open source has become a standard for software development. Sharing code freely and publicly has made it drastically cheaper and easier to build software —  **_and [tech innovation is surging as a result](https://www.fordfoundation.org/media/2976/roads-and-bridges-the-unseen-labor-behind-our-digital-infrastructure.pdf)._**

>> As the **_status quo_** for code collaboration, these platforms also host the largest repositories of open source development made up of not just code, but issues, pull requests, reviews, and comments. Even the social relationships—stars, likes, follows—exist solely within these platforms.

**Incumbents in the current phrasing does not make sense as a noun or adjective. Let's just go back to status quo**

>> **_These platforms, however, are owned by corporations._** They are subject to corporate law and have the right to define their terms of services. They can implement user bans—like those currently in place against Iranian, Syrian, and Crimean GitHub accounts in response to pressure from the U.S. government. They are vulnerable to censorship as well as capitalist ends, which are often misaligned with the goals of free and open source communities

**Comma has to be here!!**

>> Alternatives to GitHub exist ranging from platforms like **_[SourceForge](https://sourceforge.net/) and [GitLab](https://about.gitlab.com/)_**, to more established methods of collaboration such as mailing lists. Platforms like Gitea or Gogs offer self-hosted and open source solutions for code collaboration that have low platform risk but leave developers in isolated environments with no access to the global network of developers. One proposed alternative is federation. Proposals such as ForgeFed and federated GitLab are a step in the right direction, but implementations are underdeveloped or lacking. In addition, federation is dependent on domain names which can and are regularly seized by governments.

>> Let’s look at hosting platforms like GitHub or GitLab in the context of this framework: they succeed by being user-friendly and accessible, but since they are centrally controlled, they are censorable, and do not prioritize user freedoms. If we look at self-hosted solutions like Gitea, Phabricator or Gogs, they are free, uncensorable, and **_user user-friendly, (remove the extra user)_** however, they are not easily accessible due to gate-keeping and isolated environments. They also present single points of failure and require internet connectivity for most interactions with the system.

>> Radicle adopts the Scuttlebutt social overlay paradigm by establishing a peer-to-peer replication layer on top of distributed version control systems, starting with git. User accounts and login **_are_** replaced by public key cryptography, hosted issue trackers are replaced by local peer replication, and the idea of a single canonical upstream is replaced by a patch-based peer-to-peer or "bazaar" model.

>> The three major themes to highlight _**are the decisions to focus on a peer-to-peer code collaboration model, to build on the underlying distributed version control system for replication, and to adopt a protocol-first approach.**_

>> The Cathedral and the Bazaar describes two approaches to free software development. The cathedral model, exemplified by projects like Emacs, makes releases open and available but keeps development exclusive to so called "individual wizards". On the other hand, the bazaar model—popularized by _**[Linus Torvalds](http://www.catb.org/~esr/writings/cathedral-bazaar/cathedral-bazaar/ar01s04.html)**_ and validated by the massive success of Linux, calls for completely open development with frequent and early releases, delegation throughout communities, and as many "eyeballs" on the code as possible. The conclusion of the essay speaks to the success of bazaar development practices in open source projects. In other words, given enough eyeballs, all bugs are shallow.

>> To complement the eventually-consistent peer-to-peer replication layer, a global registry that holds canonical project metadata is also desirable. This can provide features difficult to address at the gossip level, such as globally unique human-readable identifiers, as well as giving the ability for projects to anchor important information about their project in an immutable and totally-ordered way—_**for example, the latest repository head.**_ This can alleviate weaknesses at the replication layer _**which are**_ unable to prove to the user that the information shown is the latest available. In essence, the registry can provide users with the trusted, auditable and shared view of the network that is otherwise lacking, delivering the same connected "social" layer that attraced developers to platforms like GitHub. Designing such a system as an open protocol is challenging, and requires a consensus-based approach which we will discuss in more depth in the future.

**NOTES:**

**Can we make the Radicle: A peer-to-peer network for code collaboration header bigger? Or in the same text as the title? Or add an emoji? 🌱 I just want it to stand out**

**Any reason for dropping the link to radicle.community? Would be good to push people there.**

Finish line! 🏁

cloudhead commented:
I've added all your edits @abbey-titcomb, thanks!

We can't make the header bigger, since it's a sub header. I'll see what it's like with an emoji.

We dropped the link to community because it's literally right below, in the footer.

cloudhead commented:
Also shouldn't it read a "p2p *protocol* for code collaboration"?
